packages/arb-avm-cpp/data_storage/src/value/machine.cpp

```diff
@@ -67,26 +60,30 @@ void serializeMachineStateKeys(const MachineStateKeys& state_data,
     marshal_uint256_t(state_data.output.log_acc, state_data_vector);
     marshal_uint256_t(state_data.output.send_count, state_data_vector);
     marshal_uint256_t(state_data.output.log_count, state_data_vector);
+    marshal_uint256_t(state_data.l1_block_number, state_data_vector);
+    marshal_uint256_t(state_data.l2_block_number, state_data_vector);
+    marshal_uint256_t(state_data.last_inbox_timestamp, state_data_vector);
 
     auto last_sideload_raw = std::numeric_limits<uint256_t>::max();
     if (state_data.output.last_sideload.has_value()) {
         last_sideload_raw = *state_data.output.last_sideload;
     }
     marshal_uint256_t(last_sideload_raw, state_data_vector);
+
+    marshal_uint256_t(state_data.static_hash, state_data_vector);
+    marshal_uint256_t(state_data.register_hash, state_data_vector);
+    marshal_uint256_t(state_data.datastack_hash, state_data_vector);
+    marshal_uint256_t(state_data.auxstack_hash, state_data_vector);
+    state_data.pc.marshal(state_data_vector);
+    state_data.err_pc.marshal(state_data_vector);
 }
 
 MachineStateKeys extractMachineStateKeys(
```

Comment:
If this function is meant to handle both checkpoint types, I think it would be cleaner and lead to more safety to have it return a `std::variant<MachineStateKeys, MachineOutput>`

Reply:
:+1: